modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/QueryTemplate.java

```diff
@@ -104,7 +103,7 @@ public ExecutionPlan map(PlanningContext ctx) {
     private List<Fragment> replace(List<Fragment> fragments, Fragment fragment, List<Fragment> replacement) {
         assert !nullOrEmpty(replacement);
 
-        Map<Long, Long> newTargets = new HashMap<>();
+        Long2LongOpenHashMap newTargets = new Long2LongOpenHashMap();
```

Comment:
`newTargets.put(remote.exchangeId(), fragment0.fragmentId());`
and
` Long newTargetId = newTargets.get(sender.exchangeId());`
use boxed values.